lib/bolt/inventory.rb

```diff
@@ -177,6 +177,22 @@ def groups_in(node_name)
     end
     private :groups_in
 
+    def self.magic_local_config(target, data)
+      if target.name == 'localhost'
+        unless data['config']['transport']
+          data['config']['transport'] = 'local'
+        end
+        if data['config']['transport'] == 'local'
```

Comment:
I think we can safely set these fields regardless of what the transport is. It doesn't matter with the way we use this now but less conditionals is usually better than more.